src/Microsoft.AspNetCore.Routing/Template/TemplateSegment.cs

```diff
@@ -4,12 +4,22 @@
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.Linq;
+using Microsoft.AspNetCore.Routing.Patterns;
 
 namespace Microsoft.AspNetCore.Routing.Template
 {
     [DebuggerDisplay("{DebuggerToString()}")]
     public class TemplateSegment
     {
+        public TemplateSegment()
+        {
+        }
+
+        public TemplateSegment(RoutePatternPathSegment other)
+        {
+            Parts = new List<TemplatePart>(other.Parts.Select(s => new TemplatePart(s)));
+        }
+
         public bool IsSimple => Parts.Count == 1;
 
         public List<TemplatePart> Parts { get; } = new List<TemplatePart>();
```

Comment:
Move this to the default `ctor()`? Avoids it being allocated twice in the other ctor.

Reply:
Sure